File: typescript/src/tools/hello-world-tool.ts
import type { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
import { z } from "zod";

export function registerHelloWorldTool(server: McpServer): void {
  server.registerTool(
    "hello_world",
    {
      description:
        "Returns a greeting message with the provided name. If name is not provided, it defaults to 'World'.",
      inputSchema: {
        name: z
          .string()
          .default("World")
          .describe("Name for greeting, for example 'Alice'. If not provided, it defaults to 'World'."),
      },
    },
    async ({ name }) => ({
      content: [
        {
          type: "text",
          text: JSON.stringify({
            Name: name,
            Message: `Hello, ${name} from typescript server!`,
          }),
        },
      ],
    })
  );
}

File: dotnet/src/McpDemo.Server/Tools/HelloWorldTool.cs
using System.ComponentModel;
using System.Text.Json;
using ModelContextProtocol.Protocol;
using ModelContextProtocol.Server;

namespace McpDemo.Server.Tools;

internal sealed class HelloWorldTool
{
    [McpServerTool(Name = "hello_world")]
    [Description("Returns a greeting message with the provided name. If name is not provided, it defaults to 'World'.")]
    public CallToolResult ExecuteAsync(
        [Description("Name for gretting, for example 'Alice'. If not provided, it defaults to 'World'.")]
        string name = "World")
    {
        return new CallToolResult
        {
            IsError = false,
            Content = new List<ContentBlock>
            {
                new TextContentBlock
                {
                    Text = JsonSerializer.Serialize(new
                    {
                        Name = name,
                        Message = $"Hello, {name} from dotnet server!"
                    })
                }
            }
        };
    }
}